### PR TITLE
aw-25: add group chat

### PR DESCRIPTION
**Pull request by canterbery**
## Describe your changes

Added api handlers for getting all chat messages and new chat messages in real time.
Added group chat page

image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-214)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/sections/chat/live/chat-message-list.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { useMemo } from 'react';
+import Box from '@mui/material/Box';
+import { Typography } from '@mui/material';
+import { format } from 'date-fns';
+import { datesFormats } from 'constants/dates-formats';
+import Scrollbar from 'components/scrollbar';
+import { ChatMessageItem } from 'components/custom';
+
+import { Message } from 'types';
```
Comment (AKutusenko) [inline src/sections/chat/live/chat-message-list.tsx]: space

File: src/sections/chat/live/chat-view.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import { useCallback, useRouter, useTranslations } from 'hooks';
+import { Button, Card, Container, Stack, Typography } from '@mui/material';
+import KeyboardArrowLeftIcon from '@mui/icons-material/KeyboardArrowLeft';
+import { Message } from 'types';
+import { CustomLink } from 'components/custom';
+import useTheme from '@mui/system/useTheme';
+import { useSendMessageMutation } from 'store/chat';
+import ChatMessageList from './chat-message-list';
+import ChatMessageInput from './chat-message-input';
+
```
Comment (AKutusenko) [inline src/sections/chat/live/chat-view.tsx:12]: imports order

**Review by AKutusenko — APPROVED** [review]

File: src/sections/chat/live/chat-view.tsx (hunk 2 above)
Comment (canterbery) [reply src/sections/chat/live/chat-view.tsx:12]: And what's wrong with this import order ?

Comment (AKutusenko) [reply src/sections/chat/live/chat-view.tsx:12]: on line 3 absolute import before importing the library

File: src/sections/chat/live/chat-message-list.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { useMemo } from 'react';
+import { Message } from 'types';
+import { Box, Typography } from '@mui/material';
```
Comment (AKutusenko) [inline src/sections/chat/live/chat-message-list.tsx:3]: imports order

**Review by AKutusenko — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**